test/unit/pdf_find_controller_spec.js

```diff
@@ -264,5 +265,327 @@ describe("pdf_find_controller", function () {
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+      selectedMatch: {
+        pageIndex: 0,
+        matchIndex: 0,
+      },
+      pageMatches: [
+        [0],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [2087],
+      ],
+      pageMatchesLength: [
+        [24],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [24],
+      ],
+    });
+  });
+
+  it("performs a search with square brackets in the query", async function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query: "[Programming Languages]",
+        caseSensitive: false,
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      selectedMatch: {
+        pageIndex: 0,
+        matchIndex: 0,
+      },
+      pageMatches: [[1501]],
+      pageMatchesLength: [[25]],
+    });
+  });
+
+  it("performs a search with parenthesis in the query", async function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query: "\t   (checks)",
+        caseSensitive: false,
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      selectedMatch: {
+        pageIndex: 1,
+        matchIndex: 0,
+      },
+      pageMatches: [[], [201]],
+      pageMatchesLength: [[], [9]],
+    });
+  });
+
+  it("performs a search with a final dot in the query", async function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    // The whitespace after the dot mustn't be matched.
+    const query = "complex applications.";
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query,
+        caseSensitive: false,
+        entireWord: false,
+        phraseSearch: true,
+        findPrevious: false,
+      },
+      matchesPerPage: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      selectedMatch: {
+        pageIndex: 0,
+        matchIndex: 0,
+      },
+      pageMatches: [[1946]],
+      pageMatchesLength: [[query.length]],
+    });
+  });
+
+  it("performs a search with a dot in the query and a missing whitespace", async function () {
+    const { eventBus, pdfFindController } = await initPdfFindController();
+
+    // The whitespace after the dot must be matched.
+    const query = "complex applications.J";
```

Comment:
Sorry, but I *really* don't understand how this makes sense!

Looking at the contents of the textLayer, there's clearly a whitespace between the dot and the following "J" which means that this query should *not* match.
What should however match is "complex applications. J" (note the added space), so what's happening here and more importantly why?

Reply:
In Acrobat (but not in Chrome), this query leads to highlight "complex applications. JavaScript" so the goal is to have something similar.
I guess that in some cases we can have a space in the text but the rendering doesn't strongly show it and so introducing of fuzziness in the query helps the user to find what he wants, as you know whitespace in pdf is a bit weird concept...

Reply:
Does the query "complex applications. J" (with a space) still work as expected?
If so that mostly alleviates my concerns (even though this adds *even more* complexity to an already very complex patch); can you please amend this unit-test to also check that case as well?